lib/bunch/binary.ex

```diff
@@ -0,0 +1,93 @@
+defmodule Bunch.Binary do
+  @moduledoc """
+  A bunch of helpers for manipulating binaries.
+  """
+
+  use Bunch
+
+  @doc """
+  Chunks given binary into parts of given size.
+
+  Remaining part is being cut off.
+
+  ## Examples
+
+      iex> <<1, 2, 3, 4, 5, 6>> |> #{inspect(__MODULE__)}.chunk_every(2)
+      [<<1, 2>>, <<3, 4>>, <<5, 6>>]
+      iex> <<1, 2, 3, 4, 5, 6, 7>> |> #{inspect(__MODULE__)}.chunk_every(2)
+      [<<1, 2>>, <<3, 4>>, <<5, 6>>]
+
+  """
+  @spec chunk_every(binary, pos_integer) :: [binary]
+  def chunk_every(binary, chunk_size) do
+    {result, _} = chunk_every_rem(binary, chunk_size)
+    result
+  end
+
+  @doc """
+  Chunks given binary into parts of given size.
+
+  Returns list of chunks and remainder.
+
+  ## Examples
+
+      iex> <<1, 2, 3, 4, 5, 6>> |> #{inspect(__MODULE__)}.chunk_every_rem(2)
+      {[<<1, 2>>, <<3, 4>>, <<5, 6>>], <<>>}
+      iex> <<1, 2, 3, 4, 5, 6, 7>> |> #{inspect(__MODULE__)}.chunk_every_rem(2)
+      {[<<1, 2>>, <<3, 4>>, <<5, 6>>], <<7>>}
+
+  """
+  @spec chunk_every_rem(binary, chunk_size :: pos_integer) :: {[binary], remainder :: binary}
+  def chunk_every_rem(binary, chunk_size) do
+    do_chunk_every_rem(binary, chunk_size)
+  end
+
+  defp do_chunk_every_rem(binary, chunk_size, acc \\ []) do
+    case binary do
+      <<chunk::binary-size(chunk_size)>> <> rest ->
+        do_chunk_every_rem(rest, chunk_size, [chunk | acc])
+
+      rest ->
+        {acc |> Enum.reverse(), rest}
+    end
+  end
+
+  @doc """
+  Cuts off the smallest possible chunk from the end of `binary`, so that the
+  size of returned binary is an integer multiply of `i`.
+
+  ## Examples
+
+      iex> import #{inspect(__MODULE__)}
+      iex> take_int_part(<<1,2,3,4,5,6,7,8>>, 3)
+      <<1,2,3,4,5,6>>
+      iex> take_int_part(<<1,2,3,4,5,6,7,8>>, 4)
+      <<1,2,3,4,5,6,7,8>>
+
+  """
+  @spec take_int_part(binary, pos_integer) :: binary
+  def take_int_part(binary, i) do
```

Comment:
I don't really like the name of this function. What would you say about `take_multiple` or `take_multiple_of`?